snake.py

```diff
@@ -0,0 +1,227 @@
+        '''
+        Extends the length of the body by adding an element at the current head position.
+        '''
+        self.body.append(self.pos)
+
+def final_screen():
+    """ Function that prints GAME OVER text
+    """
+    global p1
+    global p2
+    text1 = font.render("GAME OVER", True, WHITE)
+    text2 = font.render("YOUR SCORE", True, WHITE)
+    text3 = font.render(str(len(p1.body)), True, WHITE)
+    text4 = font.render(str(len(p2.body)), True, WHITE)
+    text5 = font.render("REPLAY?", True, WHITE)
+    text6 = font.render("Y/N", True, RED)
+
+    screen.blit(text1, (180, screen_size[1]/2-150))
+    screen.blit(text2, (170, screen_size[1]/2-50))
+    screen.blit(text3, (400, screen_size[1]/2+50))
+    screen.blit(text4, (200, screen_size[1]/2+50))
+    screen.blit(text5, (170, screen_size[1]/2+150))
+    screen.blit(text6, (400, screen_size[1]/2+150))
+    pygame.display.update()
+    
+    running = True
+    while running:
+        for event in pygame.event.get():
+            if event.type == KEYDOWN:
+                if event.key == K_y : 
+                    running = False
+
+                    ### Adds player(s)
+                    p1 = Player((360,240),STAY)
+                    p2 = Player((280,240),STAY)
+                    p2.body_color = GREEN
+                    p2.head_color = DARK_GREEN
+
+                    ### Position of fruit
+                    x_fruit = random.randrange(20, 620, 20)
+                    y_fruit = random.randrange(20, 460, 20)
+                    fruit = (x_fruit, y_fruit)
+
+                    ### Set game speed
+                    speed = 3
+
+                if event.key == K_n :
+                    print('game ended')
+                    pygame.quit()
+                    sys.exit()
+                    pygame.display.update()
+
+            # Quit if window closed
+            if event.type == QUIT:
+                print('game ended')
+                pygame.quit()
+                sys.exit()
+                pygame.display.update()
+
+### Screen
+screen_size = (640, 480)
+screen_center = (int(screen_size[0]/2),int(screen_size[1]/2))
+
+### Initiallizes the game environment
+pygame.init()
+screen = pygame.display.set_mode(screen_size)
+pygame.display.set_caption('Head Movement')
+clock = pygame.time.Clock()
+
+### Directions
+STAY = (0,0)
+UP = (0,-1)
+DOWN = (0,1)
+LEFT = (-1,0)
+RIGHT = (1,0)
+
+### Colors
+BLACK = (0,0,0)
+GREEN = (0,180,0)
+WHITE = (255,255,255)
+RED = (255, 0, 0)
+DARK_GREEN = (0,100,0)
+
+### Adds player(s)
+p1 = Player((360,240),STAY)
+p2 = Player((280,240),STAY)
+p2.body_color = GREEN
+p2.head_color = DARK_GREEN
+
+### Position of fruit
+x_fruit = random.randrange(20, 620, 20)
+y_fruit = random.randrange(20, 460, 20)
+fruit = (x_fruit, y_fruit)
+
+### Adding font and size for text in game
+font = pygame.font.SysFont("Lucida Sans Typewriter", 50)
+
+### Setting initial speed
+speed = 3
+
+### Runtime script
+running = True
+while running:
+
```

Comment:
It might make more sense to put this main loop component in a separate file named "main.py" because the name Snake.py sounds like that is should only contain what defines the Snake object not the entire game.